src/main.c

```diff
@@ -142,7 +133,15 @@ static gboolean install_start(int argc, char **argv)
 	g_main_loop_run(r_loop);
 out:
 	g_clear_pointer(&r_loop, g_main_loop_unref);
+
+	g_signal_handlers_disconnect_by_data(installer, args);
 	g_clear_pointer(&installer, g_object_unref);
+
+	g_message("installing %s done: %d", args->name, args->status_result);
```

Comment:
move this directly after the g_main_loop_run and set r_exit_status early in the function